lib/clients/websocket.js

```diff
@@ -48,28 +48,68 @@ class WebsocketClient extends EventEmitter {
     this.socket = null;
   }
 
-  onOpen() {
-    this.emit('open');
-
+  sendSubscription(type, productIDs, channels) {
     const subscribeMessage = {
-      type: 'subscribe',
-      product_ids: this.productIDs,
-      channels: this.channels,
+      type: type,
+      channels: [...channels],
     };
 
+    if (productIDs) {
+      subscribeMessage.product_ids = [...productIDs];
+    }
+
     // Add Signature
     if (this.auth.secret) {
-      let sig = signRequest(
-        this.auth,
-        'GET',
-        this.channels ? '/users/self/verify' : '/users/self'
-      );
+      let sig = signRequest(this.auth, 'GET', '/users/self/verify');
       Object.assign(subscribeMessage, sig);
     }
 
     this.socket.send(JSON.stringify(subscribeMessage));
   }
 
+  subscribeProduct(...productIDs) {
+    for (let p of productIDs) {
+      this.productIDs.add(p);
+    }
```

Comment:
This might be my personal preference, but I much prefer `Array.prototype.forEach()` usage patterns. For example:


```js
productIDs.forEach(p => this.productIDS.add(p);
```